modules/home-manager.nix

```diff
@@ -1,13 +1,20 @@
-{ config, lib, pkgs, osConfig, ... }:
+{ config, lib, pkgs, ... }@args:
 let
   cfg = config.services.flatpak;
   installation = "user";
 in
 {
 
-  options.services.flatpak = import ./options.nix { inherit cfg lib pkgs; };
+  options.services.flatpak = (import ./options.nix { inherit cfg lib pkgs; })
+  // {
+    enable = with lib; mkOption {
+      type = types.bool;
+      default = args.osConfig.services.flatpak.enable or false;
```

Comment:
Nice. I wonder if we could move this block to `options.nix`, but that's not a blocker for merging.

I think the `or false` clause can be omitted here?
`osConfig.services.flatpak.enable` is declared via a `mkEnableOption`  https://github.com/NixOS/nixpkgs/blob/70ccb0dec2983e576833de94fe35081dd5d60b5b/nixos/modules/services/desktops/flatpak.nix#L17C16-L17C30

that defaults to `false` https://github.com/NixOS/nixpkgs/blob/468356287fe4ce68f58fa5cf16e8ed6fc40a1a7f/lib/options.nix#L98

Reply:
> I wonder if we could move this block to options.nix, but that's not a blocker for merging.

I think adding it there would add it to both the home manager and NixOS module. However, `nixpkgs` already has `services.flatpak.enable`, and it would clash.

To be fair, in the NixOS module, I wouldn't add the `nix-flatpak` options directly to `services.flatpak`, but to a submodule. That is, in `modules/nixos.nix`, I'd have `options.services.flatpak.nix-flatpak = import ./options.nix`. Then, if enabled, the module would assert that `services.flatpak.enable` is also true. Then there would be no clashes and a separate enable toggle for this module. 

That said, that would be a breaking change, so I'm hesitant to do such a thing directly in this PR.

> I think the or false clause can be omitted here? `osConfig.services.flatpak.enable` is declared via a mkEnableOption

I don't follow - the main problem currently is that there is no `osConfig` when home-manager is standalone.

Reply:
> To be fair, in the NixOS module, I wouldn't add the nix-flatpak options directly to services.flatpak, but to a submodule

This would probably be the best way to do it. I've been a bit lazy to piggy back the existing NixOs module, and I'm aware that other clashes that might happen in the future.

This is likely to change as the module develops. 

> Then, we can have an enable option that's separate from services.flatpak.enable. 

I started out with this approach, but I did not like what the interface I came up with looked like. 

An alternative I've thought of would be to toggle `services.flatpak.enable` when `packages` are declared (either as an empty list, or list of attrs). 

> I don't follow - the main problem currently is that there is no osConfig when home-manager is standalone.

Yeah, no. My question did not make sense. Sorry.